### ISSUE TITLE
prevents startup if loaded in .Rprofile

### ISSUE THREAD

**Issue by katrinleinweber** [hadley/strict#5]
Is there a way to load `strict` automatically, but not _immediately_ at startup?

`library(strict)` in `.Rprofile` (alone) causes a blank RStudio (v1.0.143) screen and:
```
ERROR r error 4 (R code execution error) [errormsg=Error : [strict]
Please use `vapply()` instead of `sapply()`.
Please see ?strict_sapply for more details
]; OCCURRED AT: rstudio::core::Error rstudio::r::exec::evaluateString(const std::string &, SEXP *, sexp::Protect *) /Users/rstudio/rstudio/src/cpp/r/RExec.cpp:302
```
Seen on macOS 10.12.5 with R 3.4.0.

**Comment by dracodoc** [comment]
Looks like some RStudio code failed to pass the strict requirement. I think using `strict` on code you have no control doesn't make sense, since many things can break and you cannot fix it.

**Comment by katrinleinweber** [comment]
Agreed, in principle. However:
1. Hence the question at the top ;-)   It would be neat to let the startup phase "wash over it" and have it "wait for" my code, so to speak.
2. "cannot fix": If I knew enough C++ and RStudio internals, I could ;-) Starting probably maybe at  https://github.com/rstudio/rstudio/blob/master/src/cpp/r/RExec.cpp#L292

**Comment by dracodoc** [comment]
Yes but I imagine there will be lots of errors you need to fix when you use more packages. Because there could be quite some **loose** usage in common R code, it could be overwhelming.

**Comment by hadley** [comment]
@dracodoc strict only affects your code, not code in packages

**Comment by dracodoc** [comment]
Oh sorry, I should look at the problem and test more before make the judgement.

**Comment by hadley** [comment]
After chatting with @jjallaire, I think the best way forward in the short-term is for strict to notice that code is being run by RStudio and turn strictness off. In the long-term, we can work towards making RStudio code strict-clean.

**Comment by katrinleinweber** [comment]
Cool! I'm glad this weird experiment sparked a fruitful discussion :-)   Also, it's good to know that the people & organisations who build the tools I enjoy using are not afraid to expose improvement potential in their own work. Thank you!

**Comment by pauschae** [comment]
I think this is related. I get th efollowing error message on top of my source window when saving a notebook.:

Error creating notebook: [strict] Multiple definitions found for 'format'. Please pick one: *Matrix::format, base::format

**Comment by michaellevy** [comment]
This is still an issue. Is there a workaround to have package development always happen in strict mode?

**Comment by katrinleinweber** [comment]
I'm still seeing the same error as originally posted in RStudio v1.1.383 (now on macOS 10.13, 17A405).

**Comment by strboul** [comment]
Same issue. Still crashes when adding `strict` to the .Rprofile (`Version 1.1.447`).
